src/lib/pipeline/ExecuteUnit.cc

```diff
@@ -161,6 +162,7 @@ void ExecuteUnit::execute(std::shared_ptr<Instruction>& uop) {
   forwardOperands_(uop->getDestinationRegisters(), uop->getResults());
 
   output_.getTailSlots()[0] = std::move(uop);
+  empty_ = true;
```

Comment:
An instruction completing execution does not mean the EU is empty. Given it has an internal pipeline (as seen in the `tick` function) you'll need to make sure that a) the internal pipeline is empty, and b) there are no instructions present in the `operationsStalled` structure too

Reply:
Taken functionality from #298